src/decorators/inject.decorator.types.ts

```diff
@@ -1,11 +1,18 @@
-interface InjectMetadata {
+import { IScope } from '../types';
+
+interface IInjectMetadata {
   identifier: string;
   propertyKey?: string;
   parameterIndex?: number;
+  options?: IInjectOptions;
 }
 
-type InjectMetadataParam = Required<Omit<InjectMetadata, 'propertyKey'>>;
+type InjectMetadataParam = Required<Omit<IInjectMetadata, 'propertyKey'>>;
+
+type InjectMetadataProperty = Required<Omit<IInjectMetadata, 'parameterIndex'>>;
 
-type InjectMetadataProperty = Required<Omit<InjectMetadata, 'parameterIndex'>>;
+export type { IInjectMetadata, InjectMetadataParam, InjectMetadataProperty };
```

Comment:
Please move this statement at the very bottom of the file, and add `IInjectOptions` along the other types and interfaces. By having only one export statement it's easier to keep track of what is exported and what not.